### PR TITLE
wrap MarkdownDocument

### PR DESCRIPTION
**Pull request by 928PJY**
Wrap `MarkdownDocument` in `OverwriteDocumentModel` with *OPathString* and *line number* to improve warning message

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Markdig.Syntax;
+
+    [Serializable]
+    public class MarkdownMetadataItemModel
+    {
+        public string OPathString { get; set; }
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs]: ~~set~~ all properties are set in ctor

File: src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace Microsoft.DocAsCode.Build.OverwriteDocuments
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Markdig.Syntax;
+
+    [Serializable]
+    public class MarkdownMetadataItemModel
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs]: Add some comments to what the model stands for?

File: src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        public string OPathString { get; set; }
+
+        public int LineNumber { get; set; }
+
+        public MarkdownDocument Value { get; set; }
+
+        public MarkdownMetadataItemModel(MarkdownPropertyModel model, string file)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs]: null check

File: src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+        private MarkdownDocument CreateDocument(List<Block> blocks, string file)
+        {
+            var result = new MarkdownDocument();
+            foreach (var block in blocks)
+            {
+                block.Parent?.Remove(block);
+                result.Add(block);
+            }
+            result.SetData("filePath", file);
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs]: also set lineNumber?

File: src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs (hunk 2 above)
Comment (vicancy) [reply src/Microsoft.DocAsCode.Build.OverwriteDocuments/Models/MarkdownMetadataItemModel.cs]: A better naming?

**Comment by superyyrrzz** [comment]
Seems MarkdownDocument has some interface to store some temp values. https://github.com/lunet-io/markdig/blob/master/src/Markdig/Syntax/IMarkdownObject.cs

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Processors/FragmentsValidationInterpreter.cs [hunk 5]
```diff
@@ -45,9 +54,9 @@ public object Interpret(BaseSchema schema, object value, IProcessContext context
                 return value;
             }
 
-            // TODO: improve error message by including line number and OPathString
+            // TODO: improve error message by including line number and OPathString for yaml code
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/Processors/FragmentsValidationInterpreter.cs]: >yaml code [](start = 88, length = 9)

YAML code block

**Comment by 928PJY** [comment]
Create new PR from my fork repo to docfx:dev #2567

**Pull request closed without merging**